frontend/src/store/use-auth-context.tsx

```diff
@@ -0,0 +1,10 @@
+"use client"
+
+import { useContext } from "react";
+import {AuthContext} from "~/store/auth-context";
+
+export default function useAuthContext() {
+  const {userLogin, setUserLogin, logout} = useContext(AuthContext)
+
+  return {login: userLogin, doLogin: setUserLogin, logout}
+}
```

Comment:
I would rethink what is to be stored in auth store and let `useAuthContext` simply work as an shortcut to `useContext(AuthContext)`. Extra remapping or adding logic that does not leave in auth provider seems a bit unnecessary.

Moreover usage of `login` and `logout` in different contexts (one is a string, another one is a function) is a bit misleading. Imo `username` fits much better to describe the field that user logged in with (even if its value was an email or a phone number).

Reply:
ok this is unnecessary